mkosi/manifest.py

```diff
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import dataclasses
+import json
+from subprocess import DEVNULL, PIPE
+from textwrap import dedent
+from typing import IO, Any, Dict, List, Optional, cast
+
+from .backend import CommandLineArguments, PackageType, run
+
+
+@dataclasses.dataclass
+class PackageManifest:
+    """A description of a package
+
+    The fields used here must match
+    https://systemd.io/COREDUMP_PACKAGE_METADATA/#well-known-keys.
+    """
+
+    type: str
+    name: str
+    version: str
+    size: int
+
+    def as_dict(self) -> Dict[str, str]:
+        return {
+            "type": self.type,
+            "name": self.name,
+            "version": self.version,
```

Comment:
soo, in an earlier version i asked why not save this as the original package name. You said that it's hard to parse and hence useful to have split out. But my next question would be: why not save it in both the full standards RPM file name *and* the split out fields?

Having the full name around might be useful for cache management and such, as we then know what file to look for on disk. Having this in both ways looks like something incredibly useful to me?